demos/embeddings/watch-your-step-cora-demo.ipynb

```diff
@@ -0,0 +1,449 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.title(\"Graph log likelihood loss vs epoch\")\n",
+    "plt.xlabel(\"Epoch\")\n",
+    "plt.ylabel(\"Graph log likelihood\")\n",
+    "plt.grid()\n",
+    "plt.plot(history.history['loss'])\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing Embeddings\n",
+    "\n",
+    "Now we use TSNE to visualize the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embeddings = np.hstack(\n",
```

Comment:
I wonder if there's a way that we can make this more automatic, so that the user can access the embeddings more directly without having to know that the right embeddings are a dense layer? Do you have any thoughts about how we could customise the model or the layers to achieve this?

Reply:
that's a good idea, I'll a `WatchYourStep.get_embeddings(self, model)` function that accepts a WyS keras model and returns the embeddings